sigtool/sigtool.c

```diff
@@ -1566,6 +1573,404 @@ static int listsigs(const struct optstruct *opts, int mode)
+        return true;
+    }
+    return false;
+}
+
+static void getFilesToScan(ScanFiles *pCtx, struct json_object *pNode)
+{
+
+    struct json_object *pType             = NULL;
+    struct json_object *pName             = NULL;
+    struct json_object *pContainedObjects = NULL;
+
+    size_t i;
+
+    const char *fileType = NULL;
+    const char *fileName = NULL;
+
+    pType = json_object_object_get(pNode, "FileType");
+    pName = json_object_object_get(pNode, "FileName");
+
+    if ((NULL == pType) || (NULL == pName)) {
+        return;
+    }
+
+    fileName = json_object_to_json_string(pName);
+    fileType = json_object_to_json_string(pType);
+
+    /*Only interested in sub-files.  The RootFileType file has already been scanned.*/
+    if (NULL == json_object_object_get(pNode, "RootFileType")) {
+        if (needToScanFile(fileType)) {
+            const char *pPath        = NULL;
+            struct json_object *pObj = NULL;
+
+            pObj = json_object_object_get(pNode, "FilePath");
+            if (NULL == pObj) {
+                /*Not sure what would cause this, or what can be done in this situation*/
+                return;
+            }
+
+            pPath = json_object_to_json_string(pObj);
+
+            if (!addScanFile(pCtx, pPath)) {
+                return;
+            }
+        }
+    }
+    pContainedObjects = json_object_object_get(pNode, "ContainedObjects");
+
+    if (NULL == pContainedObjects) {
+        return;
+    }
+
+    for (i = 0; i < json_object_array_length(pContainedObjects); i++) {
+        struct json_object *pObj = NULL;
+        pObj                     = json_object_array_get_idx(pContainedObjects, i);
+        if (NULL == pObj) {
+            /*Not sure what would cause this*/
+            continue;
+        }
+
+        getFilesToScan(pCtx, pObj);
+    }
+}
+
+static int filePropsCallback(const char *jsonStr, int rc, void *context)
+{
+    struct json_object *pRoot = NULL;
+    ScanFiles *pCtx           = (ScanFiles *)context;
+
+    UNUSEDPARAM(rc);
+
+    if (NULL == jsonStr) {
+        logg(LOGG_ERROR, "NULL jsonStr passed into the file properties callback function.\n");
+        goto done;
+    }
+
+    pRoot = json_tokener_parse(jsonStr);
+    if (NULL == pRoot) {
+        logg(LOGG_ERROR, "Invalid jsonStr passed into the file properties callback function.\n");
+        goto done;
+    }
+
+    getFilesToScan(pCtx, pRoot);
+
+done:
+    return 0;
+}
+
+static int vba_callback(const unsigned char *const metaString, const size_t metaStringLen, void *context)
+{
+
+    size_t i = 0;
+
+    UNUSEDPARAM(context);
+
+    if (NULL == metaString) {
+        return -1;
+    }
+
+    for (i = 0; i < metaStringLen; i++) {
+#ifndef _WIN32
+        if ('\r' == metaString[i]) {
+            continue;
+        }
+#endif
+        printf("%c", metaString[i]);
+    }
+    printf("\n");
+
+    return 0;
+}
+
+static bool setTempDir(struct cl_engine *engine, const struct optstruct *opts)
```

Comment:
I wonder if there are other features in sigtool that could make use of setting the temp directory like this, and also the option to leaving temp files.